v2/devdoc/log_context_property_basic_types_requirements.md

```diff
@@ -82,4 +82,339 @@ typedef LOG_CONTEXT_PROPERTY_TYPE (*LOG_CONTEXT_PROPERTY_TYPE_GET_TYPE)(void);
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_018: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).copy` shall succeed and return 0. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).free
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_COPY)(void* value);
+```
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_019: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).free` shall return. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).get_type
+
+```c
+typedef LOG_CONTEXT_PROPERTY_TYPE (*LOG_CONTEXT_PROPERTY_TYPE_GET_TYPE)(void);
+```
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_020: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).get_type` shall returns the property type `LOG_CONTEXT_PROPERTY_TYPE_uint64_t`. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int32_t).to_string
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_TO_STRING)(void* property_value, char* buffer, size_t buffer_length);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_IMPL(int32_t).to_string` produces the string representation of an `int32_t`.
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_021: [** If `property_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int32_t).to_string` shall fail and return -1. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_022: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int32_t).to_string` shall call `snprintf` with `buffer`, `buffer_length` and format string `PRId32` and pass in the values list the `int32_t` value pointed to be `property_value`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_023: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int32_t).to_string` shall succeed and return the result of `snprintf`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_024: [** If any error is encountered, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int32_t).to_string` shall fail and return -1. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int32_t).copy
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_COPY)(void* dst_value, void* src_value);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_IMPL(int32_t).copy` copies the `int32_t` value from the address pointed by `src_value` to `dst_value`.
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_025: [** If `src_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int32_t).copy` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_026: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int32_t).copy` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_027: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int32_t).copy` shall copy the bytes of the `int32_t` value from the address pointed by `src_value` to `dst_value`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_028: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int32_t).copy` shall succeed and return 0. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int32_t).free
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_COPY)(void* value);
```

Comment:
>LOG_CONTEXT_PROPERTY_TYPE_COPY [](http://example.com/codeflow?start=14&length=30)

copy&Pasta detected.

Reply:
Seems like a case of here and everywhere

Reply:
Fixed.